### PR TITLE
[MAINTENANCE] Make `dataset_name` a parameter for Expectations tests or create name from `Expectation` name, which ensures only limited number of tables created.

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What does this PR do? 
* Ensures only a limited number of `temp_tables` are created by our Expectation tests, which will allow `docs-integration` to become blocking, since we will no longer be "accumulating" temp tables as a result of running our test pipelines / gallery. 
* Adds `generate_table_name_with_expectation` which generates a table name from the Expectation, so that the **same** tables will be generated whether we run the gallery or Expectation tests. 
* Adds unit tests for `generate_table_name_with_expectation` method. 
* Adds `dataset_name` for Core Expectations under test. 

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/core/expectation_diagnostics/expectation_test_data_cases.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ class ExpectationTestDataCases(SerializableDictDot):
     """Pairs a test dataset and a list of test cases to execute against that data."""
 
     data: TestData
+    dataset_name: str
```
Comment (Shinnnyshinshin) [inline great_expectations/core/expectation_diagnostics/expectation_test_data_cases.py:100]: Allows `dataset_name` to be passed in as Expectations test case

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -2729,6 +2746,22 @@ def generate_test_table_name(
     return table_name
 
 
+def generate_test_table_name_with_expectation_name(
+    dataset: dict, expectation_type: str, index: int, sub_index: int | None = None
+) -> str:
```
Comment (anthonyburdi) [inline great_expectations/self_check/util.py:2751]: Please add a docstring :)

File: tests/test_the_utils_in_self_check_utils.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+from great_expectations.self_check.util import (
+    generate_test_table_name_with_expectation_name,
+)
+
+
+@pytest.mark.parametrize(
+    "dataset,expectation_name,index,sub_index,expected_output",
+    [
+        (
```
Comment (anthonyburdi) [inline tests/test_the_utils_in_self_check_utils.py]: Consider pytest.param(id="some helpful name") to name the cases

**Review by anthonyburdi — APPROVED** [review]
Thank you for this work!! Approved with two small suggestsions.

File: tests/test_the_utils_in_self_check_utils.py (hunk 3 above)
Comment (Shinnnyshinshin) [reply tests/test_the_utils_in_self_check_utils.py]: done!

File: great_expectations/self_check/util.py (hunk 2 above)
Comment (Shinnnyshinshin) [reply great_expectations/self_check/util.py:2751]: done!

**Pull request merged**